scipy/sparse/_sputils.py

```diff
@@ -126,11 +126,12 @@ def getdtype(dtype, a=None, default=None):
                 raise TypeError("could not interpret data type") from e
     else:
         newdtype = np.dtype(dtype)
-        if newdtype == np.object_:
-            raise ValueError(
-                "object dtype is not supported by sparse matrices"
-            )
 
+    if newdtype not in supported_dtypes:
+        supported_dtypes_fmt = ", ".join([t.__name__ for t in supported_dtypes])
```

Comment:
you can drop the square brackets here: `", ".join(t.__name__ for t in supported_dtypes)`

Reply:
Fixed this, thanks!